src/libsync/owncloudpropagator.h

```diff
@@ -214,6 +214,16 @@ class PropagatorCompositeJob : public PropagatorJob
     {
         foreach (PropagatorJob *j, _runningJobs)
             j->abort();
+
+        // Give running jobs max 5 seconds to finish on their own, otherwise
+        // continue (that might result in conflicts owncloud/client/issues/5949)
+        int retriesNo = 5;
+        for(int i = 0; i < retriesNo ; i++) {
+            if (_runningJobs.size() == 0) {
+                break;
+            }
+            Utility::sleep(1);
```

Comment:
but that's not acceptable. You can't let the main thread sleep like this.
You need to find a better way to wait that the job is actually finished (by listening to the finished signal or so)